src/WinRT.Runtime/Projections/ICustomPropertyProvider.net5.cs

```diff
@@ -289,25 +390,47 @@ static unsafe ManagedCustomPropertyProviderVftbl()
 
 
         [UnmanagedCallersOnly]
-
         private static unsafe int Do_Abi_GetCustomProperty_0(IntPtr thisPtr, IntPtr name, IntPtr* result)
         {
             global::Microsoft.UI.Xaml.Data.ICustomProperty __result = default;
             try
             {
                 string _name = MarshalString.FromAbi(name);
-                object target = global::WinRT.ComWrappersSupport.FindObject<object>(thisPtr);
-                PropertyInfo propertyInfo = target.GetType().GetProperty(
-                     _name,
-                     BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public);
-
-                if (propertyInfo is object)
-                {
-                    __result = new ManagedCustomProperty(propertyInfo);
-                }
-                
-                *result = MarshalInterface<global::Microsoft.UI.Xaml.Data.ICustomProperty>.FromManaged(__result);
-
+                object target = global::WinRT.ComWrappersSupport.FindObject<object>(thisPtr);
+
+                if (target is global::Microsoft.UI.Xaml.Data.IBindableCustomPropertyImplementation bindableCustomPropertyImplementation)
+                {
+                    __result = bindableCustomPropertyImplementation.GetProperty(_name);
+                    *result = MarshalInterface<global::Microsoft.UI.Xaml.Data.ICustomProperty>.FromManaged(__result);
+                    return 0;
+                }
+
+                if (!RuntimeFeature.IsDynamicCodeCompiled)
+                {
+                    throw new NotSupportedException(
+                        $"ICustomProperty support used by XAML binding for '{target.GetType()}' requires the type to marked with 'WinRT.BindableCustomPropertyAttribute'. " +

```

Comment:
any plans to document this in greater detail, like with an md on the repo?  Xaml Compiler could then point errors/warnings to that page.

Reply:
I can get our aot.md updated to mention this in a follow up PR.